### ISSUE TITLE
Multiple warnings when creating Tendril object

### ISSUE THREAD

**Issue by stefanoborini** [Karpefors/Tendril#47]
This has been present since the initial commit (0.7, the old 1.6)

```
> mydata = Tendril(
+     mydata = TendrilData,
+     rotations = Rotations,
+     AEfreqTreshold = 9,
+     Tag = "Comment",
+     Treatments = c("placebo", "active"),
+     Unique.Subject.Identifier = "subjid",
+     Terms = "ae",
+     Treat = "treatment",
+     StartDay = "day",
+     SubjList = SubjList,
+     SubjList.subject = "subjid",
+     SubjList.treatment = "treatment"
+ )
There were 50 or more warnings (use warnings() to see the first 50)
> 
> warnings()
Warning messages:
1: In prop.test(c(m[ae1], m[ae2]), c(m[n1], m[n2])) :
  Chi-squared approximation may be incorrect
2: In prop.test(c(m[ae1], m[ae2]), c(m[n1], m[n2])) :
  Chi-squared approximation may be incorrect
3: In prop.test(c(m[ae1], m[ae2]), c(m[n1], m[n2])) :
  Chi-squared approximation may be incorrect
```

Is this expected?

**Comment by mark-edmondson-jones** [comment]
This is due to low counts for some AEs.  There is a flag suppress_warnings which is given to Tendril (default=FALSE) and passed to Tendril.stat.  This should stop these being displayed.
Martin: Would it be preferable to use the Fisher Exact test in cases with low counts (at least one expected less than 5)?  This would prevent this problem, but with the loss of some consistency for different AEs.  Alternatively the Fisher Exact test could be used in all cases?  Is there any reason to default to prop.test for colouring and only allow Fisher as an option?

**Comment by mark-edmondson-jones** [comment]
@Karpefors : Are you happy for the warnings arising from prop.test to be left (unless supressed), or would you prefer to use Fisher exact in these, or all, cases?

**Comment by Karpefors** [comment]
So fisher will not give warnings for small samples. That sounds good, but I saw a comment that it might be slower. That might be a problem for the permutation analysis... An alternative would be to select method in the call.

**Comment by mark-edmondson-jones** [comment]
Hi Martin - No It won't, but there is a point about consistency, if you run Fisher for some and chisq for others.  Running Fisher for all would be slower if you have high counts, although I'm not sure how much of a problem this would be.  Also - I don't think the stats are calculated for the permutations, only for the actual tendrils themselves, as far as I can see.

**Comment by Karpefors** [comment]
OK. If the stat calculations are not being done for permutations, that is good because they are not used. In the JAMIA paper I have mentioned that the coloring comes from a chisq test. But if we think a fisher is better we should change.

**Comment by stefanoborini** [comment]
@Karpefors After reviewing the discussion, I think that the best option is as you said: a parameter in the function with a default to the slower but more accurate option fisher. Is this ok?

**Comment by Karpefors** [comment]
Hi, yes it is ok.
/Martin

**Comment by stefanoborini** [comment]
After additional investigation, I found the following: in Tendril.stat, we always compute both the chi squared and fisher:

```
  test <- function(m) stats::prop.test(c(m[ae1], m[ae2]),c(m[n1], m[n2]))$p.value
  f.test <- function(m) stats::fisher.test(matrix(c(m[ae1], m[ae2], m[n1]-m[ae1], m[n2]-m[ae2]),2))$p.value
<...>
  for(i in unique(CountAE.wide$Terms)) {
    idx <- CountAE.wide$Terms==i
<...>
    if (suppress_warnings){
      CountAE.wide[idx, ]$p <- suppressWarnings(apply(m, 1, test))
    } else {
      CountAE.wide[idx, ]$p <- apply(m, 1, test)
    }
    CountAE.wide[idx, ]$p.adj <- stats::p.adjust(CountAE.wide[CountAE.wide$Terms==i,]$p, "fdr")
    CountAE.wide[idx,]$fish <- apply(m, 1, f.test)
```

so, there's not a lot to choose from. The warning is the correct behavior. The user has some data. The evaluation is able to provide a result, but it informs the user that data computed with chi squared are unreliable. You can still plot them and the selection is done when you use plot() (parameter coloring) but we should not prevent the user to have those numbers.

If we add the option to Tendril(), the only thing we could do is to prevent those data to be computed so that the warning is not displayed, but it would not provide any advantage, and it would introduce a situation where the same option need to be specified twice: at Tendril object creation, and when we plot it, which is an interface blunder.

So the eventual decision is: remove the choice from the plot() function, move it to Tendril(), and do not write the data, but it would still require the warning logic: what if the user _does_ ask to compute the chi squared value when calling Tendril(), and it's too unreliable? Besides, coloring accepts more options. How should the interface be?

In other words, moving it would solve nothing. The user still has the potential to show unreliable data, and the best option is the current behavior.

I strongly recommend against smart switching (use one algo or another depending on the input data). This leads to unreliable/unexpected behavior that is best avoided.

Bottom line: close and leave as is. The behavior is correct.

**Comment by Karpefors** [comment]
Yes, that sounds reasonable. Keep as is.

(FYI. The calculations are done cumulatively up to each event and therefore even with a huge dataset the prop.test will give warnings in the first rounds until enough data has accumulated.)
